File: apps/api-backend/app/services/auth_service.py
# Source: design.md TASK-B2-002 -- Auth service
"""Authentication service: register, login, refresh, logout."""

from typing import Any, Dict

from sqlalchemy.ext.asyncio import AsyncSession

from app.repositories.user_repo import UserRepository


class AuthService:
    def __init__(self, session: AsyncSession):
        self.session = session
        self.user_repo = UserRepository(session)

    async def register(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """Register a new user account.

        - Validate unique email/phone
        - Hash password
        - Create User + assign 'consumer' role
        - Return JWT pair
        """
        # TODO: implement register from T039
        pass

    async def login(self, email: str, password: str) -> Dict[str, Any]:
        """Authenticate user and return JWT pair.

        - Verify credentials (AUTH_001 on failure)
        - Check is_banned (AUTH_003 if banned)
        - Check is_active
        - Return access_token + refresh_token
        """
        # TODO: implement login from T039
        pass

    async def refresh(self, refresh_token: str) -> Dict[str, Any]:
        """Refresh an access token using a valid refresh token."""
        # TODO: implement refresh from T039
        pass

    async def logout(self, refresh_token: str) -> None:
        """Invalidate a refresh token."""
        # TODO: implement logout from T039
        pass

File: apps/mobile-app/src/features/auth/services/authService.ts
// Source: design.md Section 4.1 -- API calls stub for auth endpoints
import { apiClient } from '../../../services/apiClient';
import { TokenResponse } from '../../../types/api';

export const authService = {
  /**
   * POST /api/v1/auth/login
   * @returns JWT token pair
   */
  async login(email: string, password: string): Promise<TokenResponse> {
    // TODO: Implement API call
    const { data } = await apiClient.post('/auth/login', { email, password });
    return data.data;
  },

  /**
   * POST /api/v1/auth/register
   * [DEFERRED]
   */
  async register(
    email: string,
    password: string,
    displayName: string,
  ): Promise<TokenResponse> {
    // TODO: Implement when registration flow is active
    const { data } = await apiClient.post('/auth/register', {
      email,
      password,
      display_name: displayName,
    });
    return data.data;
  },

  /**
   * POST /api/v1/auth/refresh
   */
  async refresh(refreshToken: string): Promise<TokenResponse> {
    // TODO: Implement token refresh
    const { data } = await apiClient.post('/auth/refresh', {
      refresh_token: refreshToken,
    });
    return data.data;
  },

  /**
   * POST /api/v1/auth/logout
   */
  async logout(): Promise<void> {
    // TODO: Implement logout
    await apiClient.post('/auth/logout');
  },
};
